### PR TITLE
Error handling for cloning an agreement

### PR DESCRIPTION
**Pull request by aditya-matukumalli**

### CODE REVIEW

File: src/routes/AgreementViewRoute.js [hunk 1]
```diff
@@ -226,21 +226,31 @@ class AgreementViewRoute extends React.Component {
...
-        history.push(`${urls.agreementEdit(id)}${location.search}`);
-      });
+    });
+
+    try {
+      const text = await response.text(); // Parse it as text
+      const data = JSON.parse(text); // Try to parse it as json
+      if (response.ok) {
+        if (data.id) history.push(`${urls.agreementEdit(data.id)}${location.search}`);
+        else throw new Error('invalidJsonError'); // when the json response body doesn't contain an id
```
Comment (doytch) [inline src/routes/AgreementViewRoute.js]: Don't use magic strings, add a new file to `constants/`. `errorTypes` perhaps? Dunno.

Comment (aditya-matukumalli) [reply src/routes/AgreementViewRoute.js]: Okay. Added them in to the constants

File: src/constants/errorTypes.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default {
+  INVALID_JSON_ERROR: 'invalidJsonError',
+  JSON_ERROR: 'jsonError'
```
Comment (doytch) [inline src/constants/errorTypes.js]: It looks like you don't use this type?

File: src/routes/AgreementViewRoute.js [hunk 3]
```diff
@@ -226,21 +227,31 @@ class AgreementViewRoute extends React.Component {
...
-    }).then(response => response.json())
-      .then(({ id }) => {
-        history.push(`${urls.agreementEdit(id)}${location.search}`);
-      });
+    });
+
+    try {
+      const text = await response.text(); // Parse it as text
+      const data = JSON.parse(text); // Try to parse it as json
+      if (response.ok) {
```
Comment (doytch) [inline src/routes/AgreementViewRoute.js]: If you're not using JSON_ERROR, you can simplify this section to just `if (response.ok && data.id) ....`

**Pull request merged**